packages/regl-worldview/src/commands/GLText.js

```diff
@@ -0,0 +1,393 @@
+// @flow
+
+import TinySDF from "@mapbox/tiny-sdf";
+import memoizeOne from "memoize-one";
+import React, { useState } from "react";
+
+import { defaultBlend, defaultDepth } from "../utils/commandUtils";
+import Command, { type CommonCommandProps } from "./Command";
+import { isColorDark, type TextMarker } from "./Text";
+
+// The GLText command renders text from a Signed Distance Field texture.
+// There are many external resources about SDFs and text rendering in WebGL, including:
+// https://steamcdn-a.akamaihd.net/apps/valve/2007/SIGGRAPH2007_AlphaTestedMagnification.pdf
+// https://blog.mapbox.com/drawing-text-with-signed-distance-fields-in-mapbox-gl-b0933af6f817
+// http://hack.chrons.me/opengl-text-rendering/
+// https://stackoverflow.com/questions/25956272/better-quality-text-in-webgl
+//
+// Approach
+// ========
+// Characters from the font are measured using a <canvas> and the SDFs are drawn into a texture up front
+// (and whenever new characters are being rendered). Then one instanced draw call is made with an instance
+// per character which reads from the corresponding place in the texture atlas.
+//
+// Possible future improvements
+// ============================
+// - Add hitmap support.
+// - Allow customization of font style, maybe highlight ranges.
+// - Add a scaleInvariant option.
+// - Consider a solid rectangular background instead of an outline. This is challenging because the
+//   instances currently overlap, so there will be z-fighting, but might be possible using the stencil buffer and multiple draw calls.
+// - Somehow support kerning and more advanced font metrics. However, the web font APIs may not
+//   provide support for this. Some font info could be generated/stored offline, possibly including the atlas.
+// - Explore multi-channel SDFs.
+
+type Props = {
+  ...CommonCommandProps,
+  children: $ReadOnlyArray<TextMarker & { billboard?: boolean }>,
+  autoBackgroundColor?: boolean,
+};
+
+type FontAtlas = {|
+  textureData: Uint8Array,
+  textureWidth: number,
+  textureHeight: number,
+  charInfo: {
+    [char: string]: {|
+      x: number,
+      y: number,
+      width: number,
+    |},
+  },
+|};
+
+// Font size used in rendering the atlas. This is independent of the `scale` of the rendered text.
+const FONT_SIZE = 40;
+const BUFFER = 10;
+const MAX_ATLAS_WIDTH = 512;
+const SDF_RADIUS = 8;
+const CUTOFF = 0.25;
+const OUTLINE_CUTOFF = 0.6;
+
+const BG_COLOR_LIGHT = Object.freeze({ r: 1, g: 1, b: 1, a: 1 });
+const BG_COLOR_DARK = Object.freeze({ r: 0, g: 0, b: 0, a: 1 });
+
+// Build a single font atlas: a texture containing all characters and position/size data for each character.
+const createMemoizedBuildAtlas = () =>
+  memoizeOne(
+    (charSet: Set<string>): FontAtlas => {
+      const tinySDF = new TinySDF(FONT_SIZE, BUFFER, SDF_RADIUS, CUTOFF, "sans-serif", "normal");
+
+      const fontStyle = `${FONT_SIZE}px sans-serif`;
```

Comment:
future optimization would be nice to allow font specification to the command (likely not on the marker level since you'd need memoized build atlases for each marker)